security/fma-generic-contracts.md

```diff
@@ -0,0 +1,147 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*
+
+- [Generic items we need to take into account for any smart contract upgrade:](#generic-items-we-need-to-take-into-account-for-any-smart-contract-upgrade)
+  - [Proxy Initialization Failure](#proxy-initialization-failure)
+  - [New Implementation Overwrites (wrong) Slots](#new-implementation-overwrites-wrong-slots)
+  - [Proxy Initialized to Wrong Values](#proxy-initialized-to-wrong-values)
+  - [Can’t Upgrade Implementation](#cant-upgrade-implementation)
+  - [Able to Reinitialize Implementation](#able-to-reinitialize-implementation)
+  - [Backwards-incompatible ABIs](#backwards-incompatible-abis)
+  - [Invalid `DisputeGameFactory.setImplementation` execution](#invalid-disputegamefactorysetimplementation-execution)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+### Generic items we need to take into account for any smart contract upgrade:
+
+#### Proxy Initialization Failure
+
+- **Description:**
+  Unable to initialize the proxy due to a logic error.
+
+- **Risk Assessment:**
+  High severity since it allows arbitrary initialization.
+  Low likelihood since the initialization is a minimal operation.
+
+- **Mitigations:**
+  Extensive unit and end-to-end testing for proxy initializations.
+  Proxy initializations are verified through simulations prior to execution.
+
+- **Detection:**
+  Call initialize(..args) or by simulating the effort.
+
+- **Recovery Path(s):**
+  Perform an upgrade to perform the initialization.
+
+
+#### New Implementation Overwrites (wrong) Slots
+
+- **Description:**
+  An upgrade to a smart contract that inadvertently overwrites the wrong storage values in proxy.
+
+- **Risk Assessment:**
+  High severity due to arbitrary slot bounds.
+  Low likelihood since this is well looked out for.
```

Comment:
To be honest we do not have sufficient checks in place for this, but it's something I am hoping to fix soon ™️

Reply:
Sounds good. Will leave as-is to make the diff with notion easier to navigate when you update.

Reply:
To clarify, the work for better automated storage slot checks is not yet scheduled or prioritized, so I'd suggest adding a mitigation to manually compare the storage layout snapshots between the last release and this release